lib/matplotlib/tests/test_colorbar.py

```diff
@@ -211,11 +211,20 @@ def test_colorbar_positioning(use_gridspec):
 
 
 def test_colorbar_single_ax_panchor_false():
-    # Just smoketesting that this doesn't crash.  Note that this differs from
-    # the tests above with panchor=False because there use_gridspec is actually
-    # ineffective: passing *ax* as lists always disable use_gridspec.
+    # Note that this differs from the tests above with panchor=False because
+    # there use_gridspec is actually ineffective: passing *ax* as lists always
+    # disables use_gridspec.
+    ax = plt.subplot(111, anchor='N')
     plt.imshow([[0, 1]])
     plt.colorbar(panchor=False)
+    assert ax.get_anchor() == 'N'
+
+
+def test_colorbar_single_ax_panchor_east():
```

Comment:
Can we test this with constrained_layout as well?

Reply:
@jklymak do you mean like [this](https://github.com/matplotlib/matplotlib/compare/751e141994696ee271d9d88343c25e95823bcea1..392d4eaccbddc8893a7e3743681085c74729c6e2)?